### PR TITLE
making 'inProgress' of LocalFileProvider work

### PR DESCRIPTION
**Pull request by hansvdam**
At least  this way you can check te operation-handle to see if the task has been taken off the queue.

### CODE REVIEW

File: Sources/LocalHelper.swift [hunk 1]
```diff
@@ -280,10 +280,9 @@ open class LocalOperationHandle: OperationHandle {
     }
     
     /// Not usable in local provider
-    open var inProgress: Bool {
-        return false
-    }
-    
+    open var inProgress: Bool = false
```
Comment (amosavian) [inline Sources/LocalHelper.swift]: Please set value in `init` method, in order to make it overridable

**Comment by amosavian** [comment]
Nice solution, please implement this in `contents(path:, completionHandler:)`, `contents(path:, offset:, length:, completionHandler:)` and `writeContents(path:, contents:, atomically:, overwrite:, completionHandler:)` methods too to have consistency.

**Comment by amosavian** [comment]
Only one thing remains and that's `CloudFileProvider` handle. That's my job unfortunately though

**Comment by hansvdam** [comment]
Ok, good luck
I read you are talking about 'PosixLocalFileProvider' that offers cancel and progress for copy operations... Are you working on that currently? 
With the standard FileManager Is see workarounds. For 'Cancel' one might try http://stackoverflow.com/questions/4433231/stop-nsfilemanager-process (Maramas's answer) and for progress we might catch LocalFileProviderManagerDelegate.fileManager(_ fileManager: FileManager, shouldCopyItemAt srcURL: URL, to dstURL: URL) and do some tricks...
My personal problem is for OSX where I want a user-friendly (cancel and progress) for copying larger directories within the same volume...
What do you think?

**Comment by amosavian** [comment]
Indeed I'd prefer to implement much needed new providers (SMB2, GoogleDrive, Box-api, SugarSync, Amazon S3, etc) for now. Instead of implementing `PosixFileProvider` which is almost same with `LocalFileProvider` except for progress.

Also, dealing with posix is not easy, especially when operation is on two different partition. Another thing makes me hesitate to implement is [APFS](https://developer.apple.com/library/content/documentation/FileManagement/Conceptual/APFS_Guide/Introduction/Introduction.html) which does copying almost instantly!

Maintenance cost for lower level api is high as Apple may introduce some breaking change anytime like [this one](https://github.com/ZipArchive/ZipArchive/issues/331).

`FSCopyObjectAsync` was a good API, but it's already deprecated and can be omited anytime, I prefer not to invest in a deprecated module as I did in `FTPFileProvider` and implemented from ground instead of relying on deprecated `CFFTPStream` api.

The only way to cancel the local job, for now, is to call `[NS]Operation.cancel()` method when `isCoordinating` is `false` . But I can't assume how much it is safe and we should implement an undo job for already done files.

**Pull request merged**